### PR TITLE
[GOVCMS-282] Updated CI configuration.

### PR DESCRIPTION
**Pull request by AlexSkrypnyk**
Simplified CI to run only single job in order to save time.

Example of CI pushing tags: https://hub.docker.com/r/govcmsdev/govcms7/tags/

govcmsdev_govcms7_-_docker_hub

### CODE REVIEW

File: .ahoy.yml [hunk 1]
```diff
@@ -6,15 +6,15 @@ ahoyapi: v2
 commands:
   up:
     usage: Build project.
-    cmd: docker-compose up -d
+    cmd: docker-compose up -d "$@"
```
Comment (AlexSkrypnyk) [inline .ahoy.yml:9]: allow to pass parameters to build, if required

File: .ahoy.yml [hunk 2]
```diff
@@ -6,15 +6,15 @@ ahoyapi: v2
...
 
   down:
     usage: Delete project.
     cmd: docker-compose down
 
   build:
     usage: Build project.
-    cmd: docker-compose up -d --build
+    cmd: docker-compose up -d --build "$@"
```
Comment (AlexSkrypnyk) [inline .ahoy.yml:17]: allow to pass parameters to build, if required

File: .circleci/config.yml [hunk 3]
```diff
@@ -1,39 +1,27 @@
...
       - run:
-          name: Composer validate
+          name: Validate composer files
           command: composer validate -d tests --ansi --strict
       - run:
-          name: Update codebase to follow CI config
+          name: Update codebase to follow CI configuration
           command: sed -i -e "/###/d" /app/docker-compose.yml
-#      - setup_remote_docker
-      - setup_remote_docker:
```
Comment (AlexSkrypnyk) [inline .circleci/config.yml:26]: removed legacy code

File: .circleci/config.yml [hunk 4]
```diff
@@ -44,52 +32,20 @@ jobs:
...
-          requires:
-            - build
-          filters:
-            tags:
-              only: /^(?:7|8)\.x\-[0-9]+\.[0-9]+(?:[A-z0-9\-])*$/
-            branches:
-              # @todo: Uncomment below and remove the line after once merged.
-              #only: /master|ci/
-              only: /master|ci|feature\/7\.x\-3\.x\-lagoon/
+            if [ "${DEPLOY_ANY_BRANCH}" != "" ] || [ "${CIRCLE_BRANCH}" == "master" ]; then
```
Comment (AlexSkrypnyk) [inline .circleci/config.yml]: `$DEPLOY_ANY_BRANCH` allows to push for any branch, which is set through CI UI (to test this PR).

File: .circleci/config.yml [hunk 5]
```diff
@@ -44,52 +32,20 @@ jobs:
...
+              echo "IMAGE_VERSION_TAG=$CIRCLE_TAG">>.env.default
```
Comment (AlexSkrypnyk) [inline .circleci/config.yml]: the push command is smart enough to know if the `IMAGE_VERSION_TAG` is empty and will not push this tag if it is not provided.

File: docker-compose.yml [hunk 6]
```diff
@@ -59,7 +59,7 @@ services:
     build:
       context: .
       dockerfile: $PWD/.docker/Dockerfile.mariadb-drupal
-    image: mariadb-drupal
+    image: ${DOCKERHUB_NAMESPACE:-govcms}/mariadb-drupal
```
Comment (AlexSkrypnyk) [inline docker-compose.yml:62]: fixed missing variable here

File: .circleci/config.yml [hunk 7]
```diff
@@ -44,37 +32,23 @@ jobs:
...
-            export $(grep -v '^#' .env.default | xargs)
-            ahoy push
-
+            if [ "${DEPLOY_ANY_BRANCH}" != "" ] || [ "${CIRCLE_BRANCH}" == "master" ]; then
+              echo "IMAGE_VERSION_TAG=$CIRCLE_TAG">>.env.default
+              docker login -u $DOCKER_USER -p $DOCKER_PASS
+              export $(grep -v '^#' .env.default | xargs)
+              ahoy push
+            fi
 workflows:
```
Comment (AlexSkrypnyk) [inline .circleci/config.yml:52]: still required for CI to support builds triggered by tags and commits.

**Review by thom8 — APPROVED** [review]

**Pull request merged**